zookeeper/source/zk_dn_client.cc

```diff
@@ -120,6 +119,38 @@ namespace zkclient{
 		return created_correctly;
 	}
 
+	bool ZkClientDn::blockDeleted(uint64_t uuid, uint64_t size_bytes) {
+		int error_code;
+		bool exists;
+
+		LOG(INFO) << "DataNode deleted a block with UUID " << std::to_string(uuid);
+		std::string id = build_datanode_id(data_node_id);
+
+		bool deleted_correctly = true;
+
+		// Delete block locations
+		if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid), exists, error_code)) {
```

Comment:
In the future, we will also want to delete the corresponding work_queue/delete/<datanode_id>/<block_id> znode in this multi-op